include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -56,6 +57,10 @@ class IExecutionProvider {
    */
   virtual AllocatorPtr GetAllocator(int id, OrtMemType mem_type) const;
 
+  virtual std::unique_ptr<onnxruntime::IDataTransfer> GetDataTransfer() const {
+    return nullptr;
+  }
+
```

Comment:
I think this needs some clear documentation saying when/why it should be overridden.